src/Avalonia.Controls/ListBoxItem.cs

```diff
@@ -64,9 +64,11 @@ protected override void OnPointerPressed(PointerPressedEventArgs e)
                 if (p.Properties.PointerUpdateKind is PointerUpdateKind.LeftButtonPressed or 
                     PointerUpdateKind.RightButtonPressed)
                 {
-                    if (p.Pointer.Type == PointerType.Mouse)
+                    if (p.Pointer.Type == PointerType.Mouse
+                        || (p.Pointer.Type == PointerType.Pen && p.Properties.IsRightButtonPressed))
```

Comment:
Do we need to update the logic in PointerReleased handler here (line 108):
>
```cs
// As we only update selection from touch/pen on pointer release, we need to raise
// the pointer event on the owner to trigger a commit.
if (e.Pointer.Type != PointerType.Mouse)
```

Reply:
It's not going to get there, since _pointerDownPoint is unset